### ISSUE TITLE
False-Positiv: SuspiciousAssignmentsInspection

### ISSUE THREAD

**Issue by voku** [kalessil/phpinspectionsea#1500]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.7 |
| Language level | PHP 7.1                       |

Current behaviour
--
```php
    /**
     * Check if $url is an correct url.
     *
     * @param string $url
     * @param bool   $disallow_localhost
     *
     * @psalm-pure
     *
     * @return bool
     */
    public static function is_url(string $url, bool $disallow_localhost = false): bool
    {
        if ($url === '') {
            return false;
        }

        // WARNING: keep this as hack protection
        if (!self::str_istarts_with_any($url, ['http://', 'https://'])) {
            return false;
        }

        // e.g. -> the server itself connect to "https://foo.localhost/phpmyadmin/...
        if ($disallow_localhost) {
            if (self::str_istarts_with_any(
                $url,
                [
                    'http://localhost',
                    'https://localhost',
                    'http://127.0.0.1',
                    'https://127.0.0.1',
                    'http://::1',
                    'https://::1',
                ]
            )) {
                return false;
            }

            $regex = '/^(?:http(?:s)?:\/\/).*?(?:\.localhost)/iu';
            /** @noinspection BypassedUrlValidationInspection */
            if (\preg_match($regex, $url)) {
                return false;
            }
        }

        // INFO: this is needed for e.g. "http://müller.de/" (internationalized domain names) and non ASCII-parameters
        /** @noinspection SuspiciousAssignmentsInspection - false-positive */
        $regex = '/^(?:http(?:s)?:\\/\\/)(?:[\p{L}0-9][\p{L}0-9_-]*(?:\\.[\p{L}0-9][\p{L}0-9_-]*))(?:\\d+)?(?:\\/\\.*)?/iu';
        /** @noinspection BypassedUrlValidationInspection */
        if (\preg_match($regex, $url)) {
            return true;
        }

        /** @noinspection BypassedUrlValidationInspection */
        return \filter_var($url, \FILTER_VALIDATE_URL) !== false;
    }
```

Expected behaviour
--
no "SuspiciousAssignmentsInspection" error

Environment details
---
PhpStorm 2019.3.2
Build #PS-193.6015.48, built on January 22, 2020
Licensed to Portable UTF-8 / Lars Moelleken
Subscription is active until February 28, 2020
For non-commercial open source development only.
Runtime version: 11.0.5+10-b520.30 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-171-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 5120M
Cores: 4
Registry: run.processes.with.pty=TRUE, ide.tooltip.initialDelay=885, ide.balloon.shadow.size=0, completion.stats.show.ml.ranking.diff=true
Non-Bundled Plugins: Key Promoter X, NEON support, String Manipulation, com.alayouni.ansiHighlight, com.andrey4623.rainbowcsv, com.dubreuia, com.intellij.ideolog, com.kalessil.phpStorm.phpInspectionsUltimate, com.upsolver.PreCommitHook, de.espend.idea.php.toolbox, de.espend.idea.php.annotation, fr.adrienbrault.idea.symfony2plugin, izhangzhihao.rainbow.brackets, lv.midiana.misc.phpstorm-plugins.deep-keys, me.artspb.idea.eval.plugin, mobi.hsz.idea.gitignore, name.kropp.intellij.makefile, net.king2500.plugins.PhpAdvancedAutoComplete, net.seesharpsoft.intellij.plugins.csv, net.vektah.codeglance, org.psliwa.idea.composer, ru.adelf.idea.dotenv, tanvd.grazi, uk.co.ben-gibson.remote.repository.mapper

**Comment by rentalhost** [comment]
Confirmed.

Simplified test case:

```php
function testCase(): void
{
    if (true) {
        $var = null;

        if (true) {
            return;
        }
    }

    $var = 'ops';
}
```
